packages/db/tests/translate/test-granules.js

```diff
@@ -700,6 +704,90 @@ test('translateApiGranuleToPostgresGranule converts API granule to Postgres', as
   );
 });
 
+test('translateApiGranuleToPostgresGranule removes null/undefined fields', async (t) => {
+  const collectionCumulusId = 1;
+  const providerCumulusId = 2;
+  const pdrCumulusId = 4;
+  const dateString = new Date().toString();
+
+  const apiGranule = {
+    cmrLink: cryptoRandomString({ length: 10 }),
+    duration: 10,
+    granuleId: cryptoRandomString({ length: 5 }),
+    collectionId: constructCollectionId('name', 'version'),
+    pdrName: 'pdr-name',
+    provider: 'provider',
+    published: false,
+    status: 'running',
+    files: [
+      {
+        bucket: 'null',
+        key: 'null',
+      },
+    ],
+    beginningDateTime: dateString,
+    createdAt: Date.now() - 200 * 1000,
+    endingDateTime: dateString,
+    error: {},
+    lastUpdateDateTime: dateString,
+    processingEndDateTime: dateString,
+    processingStartDateTime: dateString,
+    productionDateTime: dateString,
+    timestamp: Date.now(),
+    timeToArchive: 0,
+    updatedAt: Date.now(),
+  };
+
+  const fakeDbClient = {};
+  const fakeCollectionPgModel = {
+    getRecordCumulusId: () => Promise.resolve(collectionCumulusId),
+  };
+  const fakeProviderPgModel = {
+    getRecordCumulusId: () => Promise.resolve(providerCumulusId),
+  };
+  const fakePdrPgModel = {
+    getRecordCumulusId: () => Promise.resolve(pdrCumulusId),
+  };
+
+  const expectedPostgresGranule = {
+    beginning_date_time: new Date(apiGranule.beginningDateTime),
+    cmr_link: apiGranule.cmrLink,
+    collection_cumulus_id: collectionCumulusId,
+    created_at: new Date(apiGranule.createdAt),
+    duration: apiGranule.duration,
+    ending_date_time: new Date(apiGranule.endingDateTime),
+    error: apiGranule.error,
+    granule_id: apiGranule.granuleId,
+    last_update_date_time: new Date(apiGranule.lastUpdateDateTime),
+    pdr_cumulus_id: pdrCumulusId,
+    processing_end_date_time: new Date(apiGranule.processingEndDateTime),
+    processing_start_date_time: new Date(apiGranule.processingStartDateTime),
+    product_volume: apiGranule.productVolume, // Undefined on apiGranule
+    production_date_time: new Date(apiGranule.productionDateTime),
+    provider_cumulus_id: providerCumulusId,
+    published: apiGranule.published,
+    query_fields: apiGranule.queryFields, // Undefined on apiGranule
```

Comment:
Is there a reason not to make these explicitly undefined?

Reply:
Ah yeah, no 😄

Updated but left the comments to be clear about what's happening: https://github.com/nasa/cumulus/pull/3082/commits/3c404fbe93c1dc25c50ed6a05b10576fb5427d53